dali/python/nvidia/dali/plugin/tf.py

```diff
@@ -144,21 +148,50 @@ def dataset_compatible_tensorflow():
   class _DALIDatasetV2(dataset_ops.DatasetSource):
     def __init__(
       self,
-      pipeline = '',
+      pipeline,
+      output_dtypes,
+      output_shapes = None,
+      *,
       batch_size = 1,
       num_threads = 4,
       device_id = 0,
       exec_separated = False,
       prefetch_queue_depth = 2,
       cpu_prefetch_queue_depth = 2,
       gpu_prefetch_queue_depth = 2,
-      shapes = [], 
-      dtypes = []):
+      dtypes=None,
+      shapes=None):
+
+      # Python will handle the case of mandatory `pipeline` and `output_dtypes` arguments
+      if dtypes is not None:
+        warnings.warn(("Use of argument `dtypes` is deprecated. Please use `output_dtypes` instead. " \
+            + "`output_dtypes` should be provided as a tuple."), Warning, stacklevel=2)
+
+      if shapes is not None:
+        if output_shapes is not None:
```

Comment:
You can something similar for `output_dtypes`.

Reply:
If you don't pass `output_dtypes` python will complain as this is an mandatory named input to a function.

Reply:
ok